[13.0] [MIG] currency_rate_update

feketemihai opened the pull request.

jgrandguillaume committed:
[ADD] First commit of the first financial modules to move in our new public branch
(lp:c2c-addons/6.1  rev 1)

vrenaville committed:
[FIX] change c2c_currency_rate_update to currency_rate_update for security
(lp:c2c-addons/6.1  rev 1.2.2)

vrenaville committed:
[MRG] merge from trunk
(lp:c2c-addons/6.1  rev 1.5.1)

yvaucher committed:
[MRG] Form trunk
(lp:c2c-addons/6.1  rev 1.1.3)

yvaucher committed:
[MRG] Form trunk
(lp:c2c-addons/6.1  rev 12)

vrenaville committed:
[FIX] currency rate update
(lp:c2c-addons/6.1  rev 1.5.15)

vrenaville committed:
[FIX] currency rate update search base on name instead of code
(lp:c2c-addons/6.1  rev 15)

Commit:
[MRG] from stable
(lp:c2c-addons/6.1  rev 1.2.14)

Commit:
[MRG] currency_rate_update fix
(lp:c2c-addons/6.1  rev 1.1.77)

yvaucher committed:
[FIX] currency_rate_update - missing dependency on account
(lp:c2c-addons/6.1  rev 17)

[FIX] currency_rate_update - change on_changes in non private methods
(lp:c2c-addons/6.1  rev 18)

yvaucher committed:
[FIX] currency_rate_update - missing dependency on account
(lp:c2c-addons/6.1  rev 1.1.96)

jgrandguillaume committed:
[TYPO] Uniformize author in Camptocamp modules
(lp:c2c-addons/6.1  rev 19)

yvaucher committed:
[FIX] contributors
(lp:c2c-addons/6.1  rev 22)

Commit:
[MRG] from 6.0 branch
(lp:c2c-addons/6.1  rev 13.1.1)

Commit:
[FIX] Merge from c2c-financial-addons 6.0. Port of account_financial_report_webkit to 6.1 :  API changes of 6.1, mainly the change of osv.memory now stored in DB, we should now have _columns, we can not rely anymore only on fields_view_get. Also shorten name files and wizard model names.
(lp:c2c-addons/6.1  rev 28)

Commit:
[MRG] from 6.1 c2c-financial-addons
(lp:c2c-addons/6.1  rev 19.1.1)

nbessi committed:
[MRG] from tunk 6.1
(lp:c2c-addons/6.1  rev 24.1.1)

Commit:
[FIX] removed unused imports
(lp:c2c-addons/6.1  rev 49)

[FIX] spelling
(lp:c2c-addons/6.1  rev 77)

Commit:
[MRG] from upstream
(lp:c2c-addons/6.1  rev 24.1.3)

Commit:
[MIGR] set installable to False on all the modules, the flag has to be reactivated when the modules are migrated

[FIX] encoding should be coding

maxime-c2c committed:
[CHG] changed icon of module currency_rate_update

eLBati committed:
[FIX] currency_rate_update views

[IMP] hiding company_id field as always used within company form

[FIX]using logging instead of netscv

[FIX] imports

[FIX] use osv.Model

[DEL] remove unused wizard directory

[FIX] loading xml files

[FIX] parameter name

[IMP] copyright headers

[IMP] logging arguments

yvaucher committed:
[IMP] remove +x rights on files

[IMP] clean spaces

[IMP] change license to AGPL

jbaudoux committed:
[FIX] imports

Commit:
Launchpad automatic translations update.

agb80 committed:
Added option to use Banxico for update USD - MXN exchange rates

Commit:
[IMP] Make base currency based on currency maked as base and not related to company currency

ddico committed:
[ADD] Currency rate updates from Bank Of Canada

Commit:
FIX replace print for orm.except_orm

Commit:
Launchpad automatic translations update.

nbessi committed:
Set all addons to uninstallable

Rename all addons to xxx_unported to be travis compliant

Revert "Rename all addons to xxx_unported to be travis compliant"

This reverts commit 7eb182744f3057dd5cb9e35faf230412aa7a2331.

[FIX] move all unported addons into __unported__ folder instead of adding _unported suffix

Fix PEP8 of module currency_rate_update in order to be able to activate flake8 testing on travis.
No refactoring has been done despite the current need

[FIX] flake8 PEP8 of module currency_rate_update

Commit:
currency_rate_update: fix wrong use of timetuple.
- remove error
"TypeError: argument must be 9-item sequence, not datetime.datetime"

currency_rate_update: use date format of openerp.tools.
- DEFAULT_SERVER_DATETIME_FORMAT and DEFAULT_SERVER_DATE_FORMAT

currency_rate_update: fix, forget to add import timedelta

currency_rate_update : mini date fix in currency_rate_update.py

[FIX] pep8 line length to 80, activate pep8 check in travis, and add noqa on scenario while waiting improvement of quality-tools

eLBati committed:
[FIX] PEP8

sbidoul committed:
[IMP] fix pylint errors

agb80 committed:
Minor fix on Banxico Updates

There were a missing return line on banxico updater that prevents
module updates rates for mexican currency updater

feketemihai committed:
[MIG] currency_rate_update: Migration to 8.0

Move directory to main folder.

Make the module instalable.

Update company to new api.

Update res currency update services to new api.

Move init file to model.

Update copyrights, colaborators will be visible to readme file.

Add readme.md, cron data for services, modify declaration of views.

Modify views.

Added abstract classes, split services in different files.

Remove self.ensure_one from @api.one.

Modify contribuitors.

Update acces rights.

Update openerp.py security file.

Re-add imports of services abstract classes.

Add dynamic import of services to pass flake8.

Remove trailing whitespaces.

Update datetime.time.min method not existent to datetime.min.time().

Calculate midnight based on time.

Update travis relative imports, warnings.

Update code according to comments.

Update CA_BoC messages.

sbidoul committed:
[FIX] pylint relative imports

[FIX] pylint redefined-outer-name

feketemihai committed:
Update strings.

Add pot file.

Remove whitespace from company.py.

Update Yahoo supported currencies, dictionary of supported currencies, fix indentation.

rschnapka committed:
initial german translation (mostly)

yvaucher committed:
Add license key in __openerp__.py

[FIX] Travis complaints

Update README.md

I chenged my data (grzegorz-openglobe)

Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

Move module icon in static/description

Update README.rst using OCA format

Add bug tracker link on README.rst

[UPD] prefix versions with 8.0

pjmcdermott committed:
[FIX] Add EUR currency option to ECB service when base currency != EUR

dreispt committed:
Small fix on description

alexis-via committed:
Add record rule for currency.rate.update.service
Workaround for bug #151
Add dedicated menu entry for currency.rate.update.service in Accounting > Config > Misc
Fix English mistake
Clean-up form view structure of currency.rate.update.service

Fix crash "TypeError: _run_currency_update() takes exactly 1 argument (2 given)"

Add info log and convert an existing info log to error log

Fix log message

company_id on currency.rate.update.service needs to have a default value

mlaitinen committed:
Fixed the Yahoo service URL. Last time the previous URL worked was on 2015-03-11.

oca-transbot committed:
OCA Transbot updated translations from Transifex

vrenaville committed:
[IMP] Port to V9

katyukha committed:
[REF] currency_rate_update: refactoring related to PR #318

Conflicts:
	currency_rate_update/services/__init__.py
	currency_rate_update/services/currency_getter.py
	currency_rate_update/services/update_service_CA_BOC.py
	currency_rate_update/services/update_service_CH_ADMIN.py
	currency_rate_update/services/update_service_ECB.py
	currency_rate_update/services/update_service_PL_NBP.py
	currency_rate_update/services/update_service_RO_BNR.py

yvaucher committed:
Fix contributors, short license headers and copyrights
There was missing a bunch of people there!

Fix translated string with extra spaces

Add a warning and a note in roadmap about Bank of canada not working

nikiwaibel committed:
[MIG] currency_rate_update: Migrated to 10.0

alexis-via committed:
[FIX+IMP] currency_rate_update: Several things:

* Update to latest OCA conventions

  Update README
  Code cleanup: remove @api.one, better error messages, remove call to MOD_NAME which was not defined any more
  Use account.menu_config_multi_currency as parent menu
* Remove res.company view
* Crash when you add currencies to "Currencies to update with this service"
* should now work in multi-company

jesusVMayor committed:
migrated currency_rate_update to 11.0

JordiBForgeFlow committed:
[FIX+IMP] currency_rate_update: fix tests + introduce compatibility with currency_rate_inverted

Commit:
Update res.config view to show auto_currency_up field and label.

Update res.config view.

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[FIX] currency_rate_update: Update error with multiple companies (#15)

Commit:
Update view, odoo compatibility.

Update version.

Update view.

Commit:
[10.0] currency_rate_update: more robust cron execution (#724)

With the current implementation, when there is a single failure in the download of currencies via cron, the field next_run is not updated, and then it the download stops forever... until someone updates next_run manually. This commit fixes that.

alexey-pelykh committed:
[MIG] currency_rate_update: Migration to 12.0

oca-travis committed:
[UPD] Update currency_rate_update.pot

alexey-pelykh committed:
[IMP] currency_rate_update: allow access company properties

alexey-pelykh committed:
[IMP] currency_rate_update: log errors

oca-travis committed:
[UPD] Update currency_rate_update.pot

alexey-pelykh committed:
[IMP] currency_rate_update: test coverage

alexey-pelykh committed:
[IMP] currency_rate_update: enhance formatting

oca-travis committed:
[UPD] Update currency_rate_update.pot

alexey-pelykh committed:
[IMP] currency_rate_update: tests to succeed on non-banking days

alexey-pelykh committed:
[IMP] currency_rate_update: use last_successful_run

oca-travis committed:
[UPD] Update currency_rate_update.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: currency-12.0/currency-12.0-currency_rate_update
Translate-URL: https://translation.odoo-community.org/projects/currency-12-0/currency-12-0-currency_rate_update/

oca-travis committed:
[UPD] Update currency_rate_update.pot

MiquelRForgeFlow committed:
[ADD] Migration scripts to v12

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: currency-12.0/currency-12.0-currency_rate_update
Translate-URL: https://translation.odoo-community.org/projects/currency-12-0/currency-12-0-currency_rate_update/

OCA-git-bot committed:
[UPD] README.rst

alexey-pelykh committed:
[FIX] currency_rate_update: no data on weekends

oca-travis committed:
[UPD] Update currency_rate_update.pot

OCA-git-bot committed:
currency_rate_update 12.0.1.1.2

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: currency-12.0/currency-12.0-currency_rate_update
Translate-URL: https://translation.odoo-community.org/projects/currency-12-0/currency-12-0-currency_rate_update/

oca-travis committed:
[UPD] Update currency_rate_update.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: currency-12.0/currency-12.0-currency_rate_update
Translate-URL: https://translation.odoo-community.org/projects/currency-12-0/currency-12-0-currency_rate_update/

Martronic-SA committed:
[12.0][IMP] currency_rate_update for non EUR companies

Fix lint, added comment when ECB and company currency differs from EUR

Add test coverage for foreign currency as base (CHF)

Update res_currency_rate_provider.xml

oca-travis committed:
[UPD] Update currency_rate_update.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: currency-12.0/currency-12.0-currency_rate_update
Translate-URL: https://translation.odoo-community.org/projects/currency-12-0/currency-12-0-currency_rate_update/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 60.5% (46 of 76 strings)

Translation: currency-12.0/currency-12.0-currency_rate_update
Translate-URL: https://translation.odoo-community.org/projects/currency-12-0/currency-12-0-currency_rate_update/hr/

Galoperd committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (76 of 76 strings)

Translation: currency-12.0/currency-12.0-currency_rate_update
Translate-URL: https://translation.odoo-community.org/projects/currency-12-0/currency-12-0-currency_rate_update/es/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 81.6% (62 of 76 strings)

Translation: currency-12.0/currency-12.0-currency_rate_update
Translate-URL: https://translation.odoo-community.org/projects/currency-12-0/currency-12-0-currency_rate_update/hr/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 84.2% (64 of 76 strings)

Translation: currency-12.0/currency-12.0-currency_rate_update
Translate-URL: https://translation.odoo-community.org/projects/currency-12-0/currency-12-0-currency_rate_update/hr/

alexey-pelykh commented on currency_rate_update/__manifest__.py:
```diff
@@ -3,29 +3,22 @@
-        'account',
+    "name": "Currency Rate Update",
+    "version": "13.0.1.0.0",
+    "author": "Camptocamp, " "Brainbean Apps, " "Odoo Community Association (OCA)",
```
The `" "` parts look really odd

feketemihai replied:
updated

alexey-pelykh replied:
hm, does not show :(

alexey-pelykh replied:
Seems autoformatting `" "` is still there

alexey-pelykh commented on currency_rate_update/models/res_currency_rate_provider.py:
```diff
@@ -156,157 +134,132 @@ def _update(self, date_from, date_to, newest_only=False):
-                        'currency': currency.name,
-                        'rate': rate,
-                    }
+                    _("Invalid rate from %(provider)s for" " %(currency)s : %(rate)s")
```
`" "`again, seems like an autoformatter

feketemihai replied:
updated

alexey-pelykh replied:
Odd, but I don't see the update :(

alexey-pelykh reviewed (changes requested):
Overall looks good, yet some `" "` in one line look not great

feketemihai commented:
@alexey-pelykh I updated the formatting, since it's coming from pre-commit.

alexey-pelykh commented on currency_rate_update/models/res_currency_rate_provider_ECB.py:
```diff
@@ -3,62 +3,95 @@
+    service = fields.Selection(selection_add=[("ECB", "European Central Bank")])
 
-    @api.multi
+    @api.model
```
Why?

alexey-pelykh replied:
It supposed to be run on instance, not on class

feketemihai replied:
@alexey-pelykh Right, reverted

alexey-pelykh commented:
@feketemihai https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0#technical-method-to-migrate-a-module-from-120-to-130-branch this suggests a bit different approach regarding that auto-formatting should've been a separate commit

MiquelRForgeFlow commented:
@feketemihai yeah, please, split that commit into two, one for new format and the other the proper migration. If not, it's very difficult to review.

oca-travis committed:
[UPD] Update currency_rate_update.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: currency-12.0/currency-12.0-currency_rate_update
Translate-URL: https://translation.odoo-community.org/projects/currency-12-0/currency-12-0-currency_rate_update/

MiquelRForgeFlow commented:
and please cherry-pick https://github.com/OCA/currency/pull/71 :)

feketemihai committed:
[IMP] currency_rate_update: black, isort

feketemihai committed:
[MIG] [13.0] Migrate currency_rate_update to 13.0

feketemihai commented:
@MiquelRForgeFlow @alexey-pelykh updated according with comments. Thanks

alexey-pelykh commented:
/ocabot merge
